homeassistant/components/pterodactyl/api.py

```diff
@@ -79,7 +79,9 @@ async def async_init(self):
 
             raise PterodactylConnectionError(error) from error
         else:
-            game_servers = paginated_response.collect()
+            game_servers = await self.hass.async_add_executor_job(
+                paginated_response.collect
+            )
```

Comment:
Can this raise?

We now have 2 executor jobs directly behind each other and switching context from the event loop to the executor is expensive, so grouping that in a single executor job would be more efficient

Reply:
Thanks for the hints.

I grouped in now into a single executor job.

And it looks like it can raise "PterodactylApiError" (which I already catch) and additionally "StopIteration", which was missing. Added that as well.